packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts

```diff
@@ -24,6 +24,11 @@ export interface ManualApprovalActionProps extends codepipeline.CommonAwsActionP
    * Any additional information that you want to include in the notification email message.
    */
   readonly additionalInformation?: string;
+
+  /**
+   * URL you want to provide to the reviewer as part of the approval request. The URL must begin with 'http://' or 'https://'.
```

Comment:
Is that condition ("must begin with 'http://' or 'https://'") checked by the CodePipeline API? Should we add some validation on our side that errors out if that is not the case?

Reply:
@skinny85  
I am not sure, i know that the AWS console does this kind of validation. Are there some common examples of this kind of validation in CDK?

Reply:
@skinny85  - Would something like this work?

```
    if (this.props.externalEntityLink) {
      const pattern = /^((http|https):\/\/)/;
      if (!pattern.test(this.props.externalEntityLink)) {
        throw Error("'externalEntityLink' - The URL must begin with 'http://' or 'https://'");
      }
    }
```

Reply:
Yep, that could work :) (+ a unit test for this behavior, of course 😃). _However_, I'd rather not add this validation unless the CodePipeline API performs it as well (I don't think we should disallows things that the API accepts). Can you verify whether the CodePipeline API fails if you provide something that does not start with 'http://' or 'https://'? Thanks!